code/modules/surgery/organs/organ.dm

```diff
@@ -323,42 +318,13 @@ var/list/organ_cache = list()
 
 	owner = target
 	processing_objects -= src
-	target.internal_organs |= src
 	affected.internal_organs |= src
-	if (!(organ_tag in target.internal_organs_by_name))
-		target.internal_organs_by_name[organ_tag] = src // In case multiple of the same type are inserted, only the first one is the primary organ
+	if (!target.get_int_organ(src))
+		target.internal_organs |= src
```

Comment:
if you are putting this behind a sanity check then at least use += to avoid the |= slowdown